### PR TITLE
Fixed Bonding tests for infiniband based interface

### PR DESCRIPTION
**Pull request by manvanthar**
Fixed Bonding tests for infiniband based interface

Signed-off-by: Manvanthara Puttashankar <manvanth@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by manvanthar** [comment]
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3783263/job.log)

job log attached

**Comment by manvanthar** [comment]
PR fixes the review comments of PR# 1334

File: io/net/bonding.py [hunk 1]
```diff
@@ -125,6 +130,20 @@ def setup_ip(self):
         self.gateway = process.system_output(
             '%s' % cmd, shell=True)
 
+    def bond_ib_conf(self, bond_name, arg1, arg2):
+        '''
+        configure slaves for IB cards
+        '''
+        cmd = 'ip link set %s up;' % (bond_name)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to bring Bond interface %s up" % bond_name)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: @manvanthar This should be `self.fail()` as @abdhaleegit suggested.

File: io/net/bonding.py [hunk 2]
```diff
@@ -125,6 +130,20 @@ def setup_ip(self):
...
+        if arg2 == "ATTACH":
+            cmd = 'ifenslave %s %s -f;' % (bond_name, arg1)
+        else:
+            cmd = 'ifenslave %s -d %s ;' % (bond_name, arg1)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to %s IB interface " % arg2)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: @manvanthar This should be `self.fail()` as @abdhaleegit suggested.

**Comment by manvanthar** [comment]
Fixed in PR # 1380

**Pull request closed without merging**